### ISSUE TITLE
knative-local-gateway causing havoc with istio sidecar

### ISSUE THREAD

**Issue by maganaluis** [kubeflow/kfserving#1659]
/kind bug

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]

I installed KNative 0.22 and KFServing 0.5.1, all works well except that now istio sidecar is broken, and I'm not able to connect to other services. 

```log
 2021-06-11T00:00:23.337646327Z 2021-06-11T00:00:23.336359Z    warning    envoy config    [external/envoy/source/common/config/grpc_subscription_impl.cc:101] gRPC config for type.googleapis.com/envoy.api.v2.RouteConfiguration rejected: Only unique values for domains are permitted. Duplicate entry of domain initial-model-predictor-default.51ed062d-d423-4be9-a3
 14-e196a7834e37.svc
 2021-06-11T00:00:23.337685027Z 2021-06-11T00:00:23.336361Z    warning    envoy config    [external/envoy/source/common/config/grpc_subscription_impl.cc:101] gRPC config for type.googleapis.com/envoy.api.v2.RouteConfiguration rejected: Only unique values for domains are permitted. Duplicate entry of domain initial-model-predictor-default.51ed062d-d423-4be9-a3
 14-e196a7834e37.svc
```

This configuration causes the issue: 

```yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: config-istio
  namespace: knative-serving
  labels:
    serving.knative.dev/release: "v0.22.0"
    networking.knative.dev/ingress-provider: istio
data:
  gateway.knative-serving.knative-ingress-gateway: "istio-ingressgateway.istio-system.svc.cluster.local"
  local-gateway.knative-serving.knative-local-gateway: "knative-local-gateway.istio-system.svc.cluster.local"
```

However if you disable the local-gateway, then you can't communicate with your inference services internally. e.g. if you try the request below you'll get a 404. That's very unfortunate because you can't get KNative Eventing this way.  

```python
requests.get("http://initial-model.51ed062d-d423-4be9-a314-e196a7834e37.svc.cluster.local/v1/models/initial-model").text
```

**What did you expect to happen:**

I expected KNative 0.22 work with KFServing 0.5.1, though this means I'll have to downgrade KNative to the Kubeflow supported version. Ideally KFserving should not have to even switch to knative-local-gateway, and for the networking issues I'd look for guidance from the KFServing team. 

**Anything else you would like to add:**

This issue likely falls within KNative, but I'd like to start the discussion here since it's most impacting to KFServing. 

**Environment:**

- Istio Version: 1.6.5
- Knative Version: 0.22
- KFServing Version: 0.5.1
- Kubeflow version: ~1.2
- Kfdef:[k8s_istio/istio_dex/gcp_basic_auth/gcp_iap/aws/aws_cognito/ibm] 
- Minikube version: N/A
- Kubernetes version: (use `kubectl version`): 1.19.X
- OS (e.g. from `/etc/os-release`): N/A

**Comment by maganaluis** [comment]
fyi the following installation works great: 

```bash
kubectl apply -f https://github.com/knative/serving/releases/download/v0.17.4/serving-crds.yaml
kubectl apply -f https://github.com/knative/serving/releases/download/v0.17.4/serving-core.yaml
kubectl apply -f https://github.com/knative/serving/releases/download/v0.17.4/serving-hpa.yaml
kubectl apply -f https://github.com/knative/net-istio/releases/download/v0.17.1/net-istio.yaml
kubectl patch configmap/config-istio \
  -n knative-serving \
  --type merge \
  -p '{"data":{"local-gateway.knative-serving.cluster-local-gateway":"cluster-local-gateway.istio-system.svc.cluster.local"}}'

kubectl apply -f https://github.com/knative/eventing/releases/download/v0.23.0/eventing-crds.yaml
kubectl apply -f https://github.com/knative/eventing/releases/download/v0.23.0/eventing-core.yaml
kubectl apply -f https://github.com/knative/eventing/releases/download/v0.23.0/in-memory-channel.yaml
kubectl apply -f https://github.com/knative/eventing/releases/download/v0.23.0/mt-channel-broker.yaml
```

**Comment by yuzisun** [comment]
@maganaluis So if I understand correctly knative 0.22 breaks KFServing when istio sidecar is injected?

**Comment by maganaluis** [comment]
@yuzisun No, I made sure the sidecar injection is disabled via the configmap in the istio-system. There is another issue I found yesterday where incoming requests from the browser, or the istio-ingressgateway are being executed twice on the inference services. See https://github.com/knative/serving/issues/11510

This networking issues are strange indeed, I think I will have to get a configuration working locally on Minikube based on the recommended versions, and upgrade our-clusters accordingly. I will update the issue then.
